File: src/main/java/yaasl/server/persistence/AuthorityRepository.java
package yaasl.server.persistence;

import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import yaasl.server.model.Authority;
import yaasl.server.model.Flight;
import yaasl.server.model.Location;

import java.util.List;

public interface AuthorityRepository extends CrudRepository<Authority, Long> {

    @Query("select authority from Authority authority where authority.authority = ?1")
    Authority findByName(String name);

}

File: src/main/kotlin/yaasl/server/persistence/AuthorityRepository.kt
package yaasl.server.persistence

import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository
import yaasl.server.model.Authority
import yaasl.server.model.Flight
import yaasl.server.model.Location

interface AuthorityRepository : CrudRepository<Authority, Long> {

    @Query("select authority from Authority authority where authority.authority = ?1")
    fun findByName(name: String): Authority

}
